test/object-store/sync/flx_migration.cpp

```diff
@@ -0,0 +1,212 @@
+#include "sync/flx_sync_harness.hpp"
+#include "sync/sync_test_utils.hpp"
+#include "util/baas_admin_api.hpp"
+
+#include <realm/object-store/impl/object_accessor_impl.hpp>
+
+#include <catch2/catch_all.hpp>
+#include <chrono>
+
+
+#if REALM_ENABLE_SYNC
+#if REALM_ENABLE_AUTH_TESTS
+
+using namespace realm;
+
+static void trigger_server_migration(const AppSession& app_session, bool switch_to_flx,
+                                     const std::shared_ptr<util::Logger>& logger = nullptr)
+{
+    auto baas_sync_service = app_session.admin_api.get_sync_service(app_session.server_app_id);
+
+    REQUIRE(app_session.admin_api.is_sync_enabled(app_session.server_app_id));
+    app_session.admin_api.migrate_to_flx(app_session.server_app_id, baas_sync_service.id, switch_to_flx);
+
+    // While the server migration is in progress, the server cannot be used - wait until the migration
+    // is complete. migrated with be populated with the 'isMigrated' value from the complete response
+    AdminAPISession::MigrationStatus status;
+    std::string last_status;
+    std::string op_stg = [switch_to_flx] {
+        if (switch_to_flx)
+            return "PBS->FLX Migration";
+        else
+            return "FLX->PBS Rollback";
+    }();
+    const int duration = 90;
+    try {
+        timed_sleeping_wait_for(
+            [&] {
+                status = app_session.admin_api.get_migration_status(app_session.server_app_id);
+                if (logger && last_status != status.statusMessage) {
+                    last_status = status.statusMessage;
+                    logger->debug("%1 status: %2", op_stg, last_status);
+                }
+                return status.errorMessage || status.complete;
+            },
+            // Query the migration status every 0.5 seconds for up to 90 seconds
+            std::chrono::seconds(duration), std::chrono::milliseconds(500));
+    }
+    catch (const std::runtime_error&) {
+        if (logger)
+            logger->debug("%1 timed out after %2 seconds", op_stg, duration);
+        REQUIRE(false);
+    }
+    if (logger) {
+        if (status.errorMessage)
+            logger->debug("%1 returned error: %2", op_stg, *status.errorMessage);
+        else
+            logger->debug("%1 complete", op_stg);
+    }
+    REQUIRE(!status.errorMessage);
+    REQUIRE(switch_to_flx == status.isMigrated);
+}
+
+// Add a set of count number of Dog objects to the realm
+static std::vector<ObjectId> fill_test_data(SyncTestFile& config, std::string partition, int start = 1, int count = 5)
```

Comment:
since we never call these without specifying the values for start/count, let's remove their default values.